lib/Backend/DecompilePipe.cpp

```diff
@@ -41,6 +41,26 @@ void Decompile::run(pipeline::ExecutionContext &EC,
     std::string CCode = decompile(Cache, *F, Model, B);
     DecompiledFunctions.insert_or_assign(Function.Entry(), std::move(CCode));
   }
+
+  // Do a sanity check on the names we are possibly emitting.
+  std::string Error = "";
+  for (const llvm::Function &Function : Module.functions()) {
+    if (B.NameBuilder.isNameForbidden(Function.getName()))
+      continue;
+
+    if (FunctionTags::QEMU.isTagOf(&Function))
+      Error += "- QEMU: " + Function.getName().str() + "\n";
+    if (FunctionTags::Helper.isTagOf(&Function))
+      Error += "- Helper: " + Function.getName().str() + "\n";
+    if (FunctionTags::OpaqueCSVValue.isTagOf(&Function))
+      Error += "- OpaqueCSVValue: " + Function.getName().str() + "\n";
+    if (FunctionTags::Exceptional.isTagOf(&Function))
+      Error += "- Exceptional: " + Function.getName().str() + "\n";
+    if (Function.isIntrinsic())
+      Error += "- Intrinsic: " + Function.getName().str() + "\n";
+  }
+  if (Error.size())
+    revng_abort(("Function names should be forbidden:\n" + Error).c_str());
 }
```

Comment:
@aleclearmind (low priority, feel free to come back to this after you finish the review)

I'm having second thoughts about including this with the PR.

I would _really_ like something like this - verifying consistency is a really good thing.

On the other hand, this is likely to cause a bunch of extra mass testing failures. What is the state of `develop` -> `master` thing? I don't think we should include this commit before that is ready (The PR should still go in without the commit, no point postponing everything over a test - perfect or not it's still miles better than the current situation where helper names are not banned at all).

Reply:
1. Can you elaborate the purpose of this piece of code? Can you make an example of name to ban?
2. Why do you think this will be a problem in mass-testing?

Reply:
(missed this somehow)

1. This just looks at all function names in the module before emission and aborts if there is at least one that is labeled as helper, while having a name name that we do not consider banned.
2. Because there's a chance I missed something that didn't show up in tests, and if there's even _one_ such name, it's going to fail a _lot_ of runs. With, most likely, an easy fix, but still high risk, low benefit.

Reply:
I'm fine with this. Let's see what happens.

Just wrap it in a function and make an assertion out of it.